src/components/Toggle/index.js

```diff
@@ -0,0 +1,36 @@
+import { Component } from 'pet-dex-utilities';
+import './index.scss';
+
+const events = ['active', 'inactive'];
+
+const html = `
+  <div class="toggle-container">
+    <input type="checkbox" id="toggle-input" class="toggle-container__input" data-select="toggle-input">
+    <label for="toggle-input" class="toggle-container__label" data-select="toggle-label"></label>
+  </div>
+`;
+
+export default function Toggle() {
+  Component.call(this, { html, events });
+
+  this.selected.get('toggle-input').addEventListener('change', (e) => {
+    const { checked } = e.target;
+
+    if (checked) {
+      this.notChecked();
+    } else {
+      this.isChecked();
+    }
+  });
+}
+
+Toggle.prototype = Object.assign(Toggle.prototype, Component.prototype, {
+  isChecked() {
+    this.selected.get('toggle-label').classList.add('checked');
+    this.emit('active');
+  },
+  notChecked() {
+    this.selected.get('toggle-label').classList.remove('checked');
+    this.emit('inactive');
+  },
```

Comment:
Esses 2 eventos não estão muito bem nomeados, `isChecked` é em convensão usado com uma pergunta tipo `is it checked?`, a questão é que esse tipo de nomeclatura é em geral usado em variaveis e não em funções, inclusive, funções devem ser nomeadas de forma imperativa conotando uma ação

Por exemplo `isChecked` pode ser alterado por `check`, em pt traduzido poderia ser uma diferença entre `estaChecado` e `checar`, o `checar` é uma ação e esse é o padrão de nomeclatura que esperamos pra funções!

O `notChecked` tambem não está muito bom, no caso é em geral bem ruim usar palavras negatórias em funções e variaveis, no caso o problema é o `not`

Acredito que o melhor a fazer nesse caso desses eventos seria ter uma função só chamada `toggle`, vou colocar um exemplo de implementação

```suggestion
  toggle(checked) {
    if (checked) this.selected.get('toggle-label').classList.add('checked');
    if (!checked) this.selected.get('toggle-label').classList.remove('checked');
    
    this.emit('toggle', checked);
  } 
```

Me fala se tiver dúvida com esse método